src/components/panel/panel.js

```diff
@@ -1194,6 +1232,24 @@ MdPanelRef.prototype._addStyles = function() {
 
 
 /**
+ * Updates the position configuration of a panel
+ * @param {MdPanelPosition} position
+ */
+MdPanelRef.prototype.updatePosition = function(position) {
+  if (!this._panelContainer) {
+    throw new Error('Panel does not exist yet. Call open() or attach().');
+  }
+
+  var self = this;
+
+  self._config['position'] = position;
+  self._$mdUtil.nextTick(function() {
```

Comment:
I guess it would work if you bind it to `this`, since `updatePosition` accessed the instance context?